docs/manual/usage.md

```diff
@@ -0,0 +1,63 @@
+# Using Home Manager {#ch-usage}
+
+Your use of Home Manager is centered around the configuration file,
+which is typically found at `~/.config/home-manager/home.nix` in the
+standard installation or `~/.config/home-manager/flake.nix` in a Nix
+flake based installation.
+
+::: note
+The default configuration used to be placed in `~/.config/nixpkgs`¸ so
+you may see references to that elsewhere. The old directory still works
+but Home Manager will print a warning message when used.
+:::
+
+This configuration file can be *built* and *activated*.
+
+Building a configuration produces a directory in the Nix store that
+contains all files and programs that should be available in your home
+directory and Nix user profile, respectively. The build step also checks
+that the configuration is valid and it will fail with an error if you,
+for example, assign a value to an option that does not exist or assign a
+value of the wrong type. Some modules also have custom assertions that
+perform more detailed, module specific, checks.
+
+Concretely, if your configuration contains
+
+``` nix
+programs.emacs.enable = "yes";
+```
+
+then building it, for example using `home-manager build`, will result in
+an error message saying something like
+
+``` console
+$ home-manager build
+error: A definition for option `programs.emacs.enable' is not of type `boolean'. Definition values:
+- In `/home/jdoe/.config/home-manager/home.nix': "yes"
+(use '--show-trace' to show detailed location information)
+```
```

Comment:
I wonder if the Markdown renderer actually supports the `console` language? For this block the generated classes are `programlisting hljs language-bash` which won't work:
![swappy-20231129_231930](https://github.com/nix-community/home-manager/assets/798147/b3f19371-a208-438f-88f2-3d7ea7a1bccb)

In the existing manual this block gets the classes `programlisting console hljs language-shell`.

Reply:
It looks like the above bad rendering is because the language used by the `console` snippets is set to `bash`, as in a bash script, not console output.

Reply:
Yeah, as I discovered elsewhere the new generator does not pass on the language tag. So the generated HTML contains

``` html
<pre class="programlisting">…</pre>
```

while the nmd generator emits

``` html
<pre class="programlisting console">…</pre>
```

which makes highlight-js pick the right syntax without having to attempt autodetection.

Reply:
@pennae The `html.py` code in nixos-render-docs contains

``` python
    def fence(self, token: Token, tokens: Sequence[Token], i: int) -> str:
        # TODO use token.info. docbook doesn't so we can't yet.
        return f'<pre class="programlisting">\n{escape(token.content)}</pre>'
```

Do you remember why the info cannot be passed on? I'm thinking something along the lines of

```
info = quoteattr(info) if info is not None else ""
return f'<pre class="programlisting{info}">\n{escape(token.content)}</pre>'
```

Reply:
@pennae I tried to use that PR but unfortunately `highlightjs` as specified in `nmd` fails to pick up the syntax highlighting altogether when using those changes. I'll see if I can modify the highlighting in nmd to work with the `<code>` blocks.

Reply:
@pennae Would it be possible to backport https://github.com/NixOS/nixpkgs/pull/271495 to nixpkgs-23.11?

Reply:
With this style override
```css
pre {
  padding: 0;
}

pre code.hljs {
  border: none;
  margin: 0;
}

it renders ok to me. The selection mask on `$` seems to work as well. However, this requires using the above pull request.

```
![shell](https://github.com/nix-community/home-manager/assets/217918/b5412586-f8cc-41fe-92a4-e5371ef960f1)
![nix](https://github.com/nix-community/home-manager/assets/217918/1fe60626-4d83-4007-9d02-be8776b12c8d)